src/modules/numbers/numbers.controller.ts

```diff
@@ -0,0 +1,41 @@
+import { Body, Controller, Get, Query, Post, UseInterceptors, CacheInterceptor } from '@nestjs/common';
+import {
+  ApiBearerAuth,
+  ApiBody,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { NumbersService } from './numbers.service';
+import { CreateNumberDto } from './dto/create-number.dto';
+import { UkefNumber } from './entities/number.entity';
+
+@ApiBearerAuth()
+@ApiTags('numbers')
+@Controller('numbers')
+export class NumbersController {
+  constructor(private readonly numberService: NumbersService) {}
+
+  @Post()
+  @ApiOperation({ summary: 'Create Number' })
+  @ApiBody({ type: [CreateNumberDto]})
+  @ApiResponse({ status: 201, description: 'Created.' })
+  async create(@Body() ukefIDRecord: CreateNumberDto[]): Promise<UkefNumber[]> {
+    const createResult = await this.numberService.create(ukefIDRecord);
+    return createResult;
+  }
+
+  @Get()
+  @UseInterceptors(CacheInterceptor)
+  @ApiOperation({ summary: 'Get information about UKEF Id' })
+  @ApiResponse({
+    status: 200,
+    description: 'The found record',
+    type: Number,
+  })
+  async findOne(@Query('typeId') typeId: number, @Query('number') ukefID: string): Promise<any> {
+    const UkefIdInfo = await this.numberService.findOne(typeId, ukefID);
```

Comment:
same as above - do we need to validate the `typeId` and `ukefID` properties

Reply:
we should keep the naming conventions consistent - camelCase
so this can be as simple as: `ukefId`

Reply:
Unfortunately query params are not validated against OpenApi spec that is defined above.
I added ParseIntPipe for type and created custom Pipe class for ukefId.
NodeJs query parameter validation examples are using Dto classes. I wanted to simplify it, but not sure how we will do query parameter validation in future.

Reply:
ok, no worries. I need to have a look at this one as well. let's leave it as it is for now, and we look into it in a separate PR